### PR TITLE
Added support for vagrant box versioning

### PR DESCRIPTION
**Pull request by alop**
As per https://docs.vagrantup.com/v2/boxes/versioning.html,
molecule will now add a `box_version` if it's specified in the
molecule.yml file.

### CODE REVIEW

File: molecule/templates/vagrantfile.j2 [hunk 1]
```diff
@@ -3,6 +3,9 @@ Vagrant.configure('2') do |config|
   {%- for platform in config.vagrant.platforms %}
     {%- if platform.name == current_platform %}
   config.vm.box = '{{ platform.box }}'
+      {%- if 'box_version' in platform %}
+  config.vm.box_version = "{{ platform.box_version|join }}"
```
Comment (abrown-sg) [inline molecule/templates/vagrantfile.j2]: Not clear what the join is doing?

**Comment by retr0h** [comment]
:100:

**Comment by abrown-sg** [comment]
Can you update https://github.com/metacloud/molecule/blob/master/molecule/templates/molecule.yml.j2 with a line (commented out is fine) that includes this feature? I try to keep that updated as the kitchen sink of all possible options.

File: molecule/templates/vagrantfile.j2 (hunk 1 above)
Comment (alop) [inline molecule/templates/vagrantfile.j2]: box_version is optional, if it exists in molecule.yml, we render it in vagrantfile

Comment (alop) [inline molecule/templates/vagrantfile.j2]: Oh, you mean about the join, Sorry, that's an artifact from previous testing, I'll yank that

**Comment by abrown-sg** [comment]
Once the update is made to molecule.yml.j2, :100:. Awesome work!

File: docs/source/providers.rst [hunk 2]
```diff
@@ -160,6 +161,7 @@ Other Notes
 
 * `private_key_path`, as with several other values, can be any valid Ruby because it will appear in the Vagrantfile that molecule will generate.
 
+* `box_version`, defaults to '=', can include an constrains like '<, >, >=, <=, ~.' as listed in the `Versioning`_ docs.
```
Comment (abrown-sg) [inline docs/source/providers.rst]: "can include an constrains" ?

**Comment by retr0h** [comment]
Looks good once the changes above are corrected :+1:

**Comment by alop** [comment]
I got them all

**Comment by abrown-sg** [comment]
:shipit:

**Pull request merged**